src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -596,6 +596,29 @@ public void MarkEventComplete(CompilationEvent compilationEvent, IEnumerable<Dia
             }
         }
 
+        /// <summary>
+        /// Marks the given event as fully analyzed for the unprocessed analyzers in the given analysisScope.
+        /// </summary>
+        public void MarkEventCompleteForUnprocessedAnalyzers(
+            CompilationEvent completedEvent,
+            AnalysisScope analysisScope,
+            PooledHashSet<DiagnosticAnalyzer> processedAnalyzers)
```

Comment:
>PooledHashSet [](start = 12, length = 13)

Minor: This parameter could be `HashSet<>` rather than `PooledHashSet<>`. Same comment for the other new methods in the class.